### PR TITLE
Google Artifact Registry image hashing support [BT-335]

### PR DESCRIPTION
**Pull request by mcovarr**

### CODE REVIEW

File: dockerHashing/src/main/scala/cromwell/docker/DockerInfoActor.scala [hunk 1]
```diff
@@ -230,15 +230,15 @@ object DockerInfoActor {
...
       c.copy(throttle = c.throttle.orElse(DockerConfiguration.instance.deprecatedGcrApiQueriesPer100Seconds))
-      new GcrRegistry(c)
+      new GoogleRegistry(c)
     }
 
     // To add a new registry, simply add it to that list
     List(
       ("dockerhub", { c: DockerRegistryConfig => new DockerHubRegistry(c) }),
-      ("gcr", gcrConstructor),
+      ("gcr", googleConstructor),
```
Comment (mcovarr) [inline dockerHashing/src/main/scala/cromwell/docker/DockerInfoActor.scala:241]: #6437 renames `gcr` to `google`

File: dockerHashing/src/test/scala/cromwell/docker/registryv2/AlibabaCloudCRRegistrySpec.scala [hunk 2]
```diff
@@ -19,7 +19,7 @@ object AlibabaCloudCRRegistrySpec {
        |enable = true
        |# How should docker hashes be looked up. Possible values are "local" and "remote"
        |# "local": Lookup hashes on the local docker daemon using the cli
-       |# "remote": Lookup hashes on docker hub and gcr
+       |# "remote": Lookup hashes on docker hub, gcr, and gar
```
Comment (kshakir) [inline dockerHashing/src/test/scala/cromwell/docker/registryv2/AlibabaCloudCRRegistrySpec.scala:22]: ToL: Of all the duplicates in the codebase of this string (maybe 4 at my rough count [[0]](https://github.com/broadinstitute/cromwell/blob/65/dockerHashing/src/test/scala/cromwell/docker/registryv2/AlibabaCloudCRRegistrySpec.scala#L22) [[1]](https://github.com/broadinstitute/cromwell/blob/65/cromwell.example.backends/cromwell.examples.conf#L241) [[2]](https://github.com/broadinstitute/cromwell/blob/65/core/src/main/resources/reference.conf#L364) [[3]](https://github.com/broadinstitute/cromwell/blame/65/docs/tutorials/Containers.md#L460))... why was this copy updated, and only this copy?

This instance of the copy-pasted `method` key isn't even used and should probably be deleted at some point. The code below that is only looking for the `alibabacloudcr` stanza.

File: dockerHashing/src/test/scala/cromwell/docker/DockerImageIdentifierSpec.scala [hunk 3]
```diff
@@ -17,6 +17,7 @@ class DockerImageIdentifierSpec extends AnyFlatSpec with CromwellTimeoutSpec wit
       ("index.docker.io/ubuntu",         Option("index.docker.io"),   None,                   "ubuntu",     "latest"),
       ("broad/cromwell/submarine",                None,               Option("broad/cromwell"), "submarine",  "latest"),
       ("gcr.io/google/slim",              Option("gcr.io"),         Option("google"),         "slim",     "latest"),
+      ("us-central1-docker.pkg.dev/google/slim",              Option("us-central1-docker.pkg.dev"),         Option("google"),         "slim",     "latest"),
```
Comment (kshakir) [inline dockerHashing/src/test/scala/cromwell/docker/DockerImageIdentifierSpec.scala:20]: ToL: The original intent of horizontal whitespace alignment seems to have been abandoned [as was foretold](https://google.github.io/styleguide/javaguide.html#s4.6.3-horizontal-alignment). Would you consider removing the extra &emsp;&emsp;&emsp;&emsp; spaces in &emsp; this &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;block of &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp; code?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,9 +1,14 @@
 # Cromwell Change Log
 
+## 66 Release Notes
```
Comment (kshakir) [inline CHANGELOG.md:3]: Execute order 66.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,9 +1,14 @@
...
+
+### Google Artifact Registry Support
+Cromwell now supports call caching when using Docker images hosted on
+[Google Artifact Registry](https://cloud.google.com/artifact-registry). 
+
 ## 65 Release Notes
 
 * An additional set of metrics relating to metadata age were added.
-* No user facing changes in Cromwell 65.
```
Comment (kshakir) [inline CHANGELOG.md:6]: If removing this line be sure to [also edit the releases page](https://github.com/broadinstitute/cromwell/releases/tag/65).

**Review by kshakir — APPROVED** [review]
One more note otherwise LGTM

File: CHANGELOG.md (hunk 5 above)
Comment (mcovarr) [reply CHANGELOG.md:6]: Done

**Review by breilly2 — APPROVED** [review]
👍   Looks reasonable to me

**Pull request merged**